pkg/apply/apply.go

```diff
@@ -0,0 +1,59 @@
+package apply
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getExistingObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstructured) (*uns.Unstructured, string, error) {
+	name := obj.GetName()
+	namespace := obj.GetNamespace()
+	if name == "" {
+		return nil, "", errors.Errorf("Object %s has no name", obj.GroupVersionKind().String())
+	}
+	gvk := obj.GroupVersionKind()
+	// used for logging and errors
+	objDesc := fmt.Sprintf("(%s) %s/%s", gvk.String(), namespace, name)
+	log.Printf("reconciling %s", objDesc)
+
+	// Get existing
+	existing := &uns.Unstructured{}
+	existing.SetGroupVersionKind(gvk)
+	err := client.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, existing)
+	return existing, objDesc, err
+}
+
+// ApplyObject applies the desired object against the apiserver,
+// merging it with any existing objects if already present.
+func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstructured) error {
+	existing, objDesc, err := getExistingObject(ctx, client, obj)
+
```

Comment:
Together with the `Owns` method, you must set:
https://github.com/operator-framework/operator-sdk/blob/latest/testdata/go/v3/memcached-operator/controllers/memcached_controller.go#L195
~~~
	// Set Memcached instance as the owner and controller
	ctrl.SetControllerReference(<owner configuration>, <daemonset>, r.Scheme)
~~~
That way, the ownerref is added to the daemonset and the ds is linked to the configuration object. Whenever the daemonset is messed with, the reconciler is triggered immediately.